cfn-resources/cfn-publishing-automation/cfn-3P-provider-register.yaml

```diff
@@ -0,0 +1,272 @@
+          typename = config['typeName']
+          if events['path'] == "root":
+            return {'typename': typename, 'runtime': runtime, 'path': '.'}
+          else:
+            return {'typename': typename, 'runtime': runtime, 'path': events['path']}
+
+      InputPayload:
+        url: '{{ Repository }}'
+        path: '{{ Path }}'
+        branchname: '{{ BranchName }}'
+    outputs:
+      - Selector: $.Payload.runtime
+        Type: String
+        Name: runtime
+      - Name: path
+        Selector: $.Payload.path
+        Type: String
+      - Name: resourcename
+        Selector: $.Payload.resourcename
+        Type: String
+      - Name: typename
+        Selector: $.Payload.typename
+        Type: String
+    description: The step reads the .rpdk file and determines what build environment to use.
+  - name: Build_Branch
+    action: 'aws:branch'
+    inputs:
+      Choices:
+        - NextStep: Go_Build_Enviroment
+          Variable: '{{Select_Language.runtime}}'
+          StringEquals: go
+    description: Creates an environment that deploys a 3rd party resource
+    nextStep: Start_Build
+  - name: Go_Build_Enviroment
+    action: 'aws:executeAwsApi'
+    inputs:
+      Service: codebuild
+      Api: CreateProject
+      artifacts:
+        type: NO_ARTIFACTS
+      description: This environment builds a 3P resource
+      environment:
+        computeType: BUILD_GENERAL1_SMALL
+        image: 'aws/codebuild/amazonlinux2-x86_64-standard:4.0'
+        type: LINUX_CONTAINER
+        privilegedMode: true
+      name: '{{ProjectName}}'
+      serviceRole: '{{ BuilderRole }}'
+      timeoutInMinutes: 480
+      source:
+        buildspec: |
+          version: 0.2
+          phases:
+            install:
+              runtime-versions:
+                python: 3.9
+                golang: 1.18
+              commands:
+                - pip install pre-commit
+                - yum update -y
+                - yum install docker -y
+            build:
+              commands:
+                - aws --version
+                - wget "https://github.com/aws/aws-sam-cli/releases/latest/download/aws-sam-cli-linux-x86_64.zip"
+                - unzip aws-sam-cli-linux-x86_64.zip -d sam-installation
+                - ./sam-installation/install
+                - pip3 install cloudformation-cli cloudformation-cli-go-plugin
+                - curl  https://load-gen-builds-peerislands.s3.amazonaws.com/mongodb-enterprise-6.0.repo > /etc/yum.repos.d/mongodb-enterprise-6.0.repo
+                - sudo yum install -y mongodb-atlas-cli
+                - export ATLAS_ORG_ID={{OrgID}}
+                - export ATLAS_PUBLIC_KEY={{PubKey}}
+                - export ATLAS_PRIVATE_KEY={{PvtKey}}
+                - export MCLI_PUBLIC_API_KEY=$ATLAS_PUBLIC_KEY
+                - export MCLI_PRIVATE_API_KEY=$ATLAS_PRIVATE_KEY
+                - export MCLI_ORG_ID=$ATLAS_ORG_ID
+                - export MCLI_OUTPUT=json
```

Comment:
```suggestion
                - export ATLAS_ORG_ID={{OrgID}}
                - export ATLAS_PUBLIC_KEY={{PubKey}}
                - export ATLAS_PRIVATE_KEY={{PvtKey}}
                - export ATLAS_OUTPUT=json
```

Reply:
some tests still use MCLI_* variables, those will need to be updated first, I'll list those out

Reply:
Resource tests using MCLI_* variables:

Cloud-provider-access
Custom-dns-config-cluster-aws
Federated-settings-org-role
Network-container
Private-endpoint-regional-mode